nvtabular/ops/categorify.py

```diff
@@ -766,11 +842,31 @@ def _encode(
             codes = cudf.DataFrame({"order": cp.arange(len(gdf))})
             for c in selection_l:
                 codes[c] = gdf[c].copy()
-        labels = codes.merge(
-            value, left_on=selection_l, right_on=selection_r, how="left"
-        ).sort_values("order")["labels"]
-        labels.fillna(na_sentinel, inplace=True)
-        labels = labels.values
+                if buckets:
+                    if c in buckets:
+                        codes[c + "_hashed"] = _hash_bucket(gdf, buckets, c)
+        # apply frequency hashing
+        if freq_threshold and buckets:
```

Comment:
Rather than 3 independent if statements - can you refactor to a 'if / elif / else'  chain?

Reply:
@benfred  I am modifying the conditions but I dont think `if / elif / else` chain would work here :/  The scenarios are a bit complicated to use `else`.